ui/src/app/workflows/components/workflow-dag/workflow-dag.tsx

```diff
@@ -178,12 +181,19 @@ export class WorkflowDag extends React.Component<WorkflowDagProps, WorkflowDagRe
         super(props);
         this.state = {
             ...this.getOptions(),
-            expandNodes: new Set()
+            expandNodes: new Set(),
+            waitForUpdate: false
         };
     }
 
+    public componentDidUpdate(prevProps: Readonly<WorkflowDagProps>, prevState: Readonly<WorkflowDagRenderOptions>, snapshot?: any): void {
+        if (prevState.waitForUpdate) {
+            this.setState({waitForUpdate: false});
+        }
+    }
+
     public render() {
-        if (!this.props.nodes) {
+        if (!this.props.nodes || this.state.waitForUpdate) {
```

Comment:
Probably want to distinguish the UX between "Loading" and "Workflow is malformed, waiting for an update"